### DISCUSSION TITLE
Using touch display (480x1920) possible?

### DISCUSSION THREAD

**Discussion by yerode** [tesla-android/issue-tracker#236] [Ideas]
First: Great Project! Thanks for your time and dedication!

I'm using a Pi4 8GB @2.1GHz with the Alcatel LTE (o2 Germany) and CarlinKit.

I am thinking about using exactly the Tesla Android Image but with an external touch display - I have a 8.8" 480x1920 (or 1920x480) IPS touch display working without issues.  Everything on that display a little smallish at the moment. ;-) Is there a possibility to change resolution and aspect ratio?

BTW: Looking forward to GPS support.

**Comment by yerode** [comment]
Update: I found a way to configure the config.txt for this specific display and the Pi 4 here: [https://gist.github.com/innovodesign/3f5775d19cb890c0aa59fbb96757bf4b](url)

Now the display shows the boot sequence with the right resolution and orientation. Good! 

But the specific resolution of "Tesla Android" for the display seems to be "hard coded"? ¯\_(ツ)_/¯ So far I wasn't able to change the Android resolution. There seems to be a way with some Apps (using ADB) but unfortunately so far no success.

Any ideas?

**Reply by mikegapinski** [reply]
The display resolution will be configurable with a HTTP request soon(and a single setprop that'll survive a reboot), I need this for my full screen frontend app. Work at progress at the moment, no ETA.

You can change the resolution after the virtual display service starts with: "add root && wm size 480x1920". You'll need to do this every reboot unless you recompile the OS. I am sure that the performance will suffer, 1920 is a lot of extra pixels to encode.

If you want an extra display don't use Tesla Android. It will never work well and the displays will be mirrored on top of the performance penalty.

**Reply by yerode** [reply]
Thanks for your reply. 👍 I understand. Then I will maybe use another android build for that purpose.

Until then I will stick with Tesla Android Project. Works well. Thanks again.

**Reply by mikegapinski** [reply]
@yerode having the external display mirrored on the Tesla display is something that might come soon(Elon time). Apparently people are installing 1080p displays in their second row for video playback for children.

Expected behavior in this case:
- Virtual display changes its aspect ratio to match the HDMI display. Touch in the browser remains functional. Pixel density configurable in the front end. 
- Audio playing via the Tesla speaker system
- Same content visible on all connected displays. Not expecting a perfect lip-sync but certainly usable.

Does this flow satisfy your use case?

**Reply by mikegapinski** [reply]
Feel free to reach out to Thomas on Twitter(sw1002)/German TFF forum. He has the crazy ideas on how to push Tesla Android to it's limits :D

**Reply by yerode** [reply]
Thanks again. But no. I just wanted to experiment with an 8.x" 480x1920 "slim" addon display next to the the 15" original display (or horizontally beneath the tesla display). I thought it would be great to have both, the tesla display with route planning incl battery conditioning and an small display with an android system next to it with waze/maps/abrp. ¯\_(ツ)_/¯ Was just an idea.
